internal/provisioner/kops_provisioner.go

```diff
@@ -251,12 +266,109 @@ func (provisioner *KopsProvisioner) CreateCluster(cluster *model.Cluster, aws aw
+	prometheusDNS := fmt.Sprintf("%s.prometheus.%s", cluster.ID, provisioner.privateDNS)
+	prometheus := helmDeployment{valuesPath: "helm-charts/prometheus_values.yaml", chartName: "stable/prometheus", namespace: "prometheus", chartDeploymentName: "prometheus-client", setArgument: fmt.Sprintf("server.ingress.hosts={%s}", prometheusDNS)}
+
+	helmDeployments := []helmDeployment{privateNginx, prometheus}
+
+	for _, value := range helmDeployments {
+		err = installHelmChart(value, logger, kops.GetKubeConfigPath())
+		if err != nil {
+			return err
+		}
+	}
+
+	// Get the new ELB internal-nginx endpoint
+	endpoint, err := getLoadBalancerEndpoint("internal-nginx", logger, kops.GetKubeConfigPath())
+	if err != nil {
+		return err
+	}
+
+	for _, app := range helmApps {
+		logger.Infof("Registering DNS for %s", app)
+		dns := fmt.Sprintf("%s.%s.%s", cluster.ID, app, provisioner.privateDNS)
+		err = aws.CreateCNAME(dns, []string{endpoint}, logger)
+		if err != nil {
+			return err
+		}
+	}
+
 	logger.WithField("name", kopsMetadata.Name).Info("Successfully created cluster")
 
 	return nil
 }
 
-// HelmSetup is used for the initial setup of Helm in cluster
+// waitForHelmRunning is used to check when Helm is ready to install charts
+func waitForHelmRunning(ctx context.Context, configPath string) error {
+	for {
+		cmd := exec.Command("helm", "ls", "--kubeconfig", configPath)
+		var out bytes.Buffer
+		cmd.Stderr = &out
+		cmd.Run()
+		if out.String() == "" {
+			return nil
+		}
+		select {
+		case <-ctx.Done():
+			return (errors.Wrap(ctx.Err(), "timed out waiting for helm to become ready"))
+		case <-time.After(5 * time.Second):
+		}
+	}
+}
+
+// getLoadBalancerEndpoint is used to get the endpoint of the internal ingress.
+func getLoadBalancerEndpoint(namespace string, logger log.FieldLogger, configPath string) (string, error) {
+	k8sClient, err := k8s.New(configPath, logger)
+	if err != nil {
+		return "", err
+	}
+	services, err := k8sClient.Clientset.CoreV1().Services(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, service := range services.Items {
+		if service.Status.LoadBalancer.Ingress != nil {
+			endpoint := service.Status.LoadBalancer.Ingress[0].Hostname
+			logger.Infof("Succesfully got LoadBalancer endpoint %s for Namespace %s", endpoint, namespace)
+			return endpoint, nil
+		}
+	}
+	return "", nil
+}
+
+// installHelmChart is used to install Helm charts.
+func installHelmChart(chart helmDeployment, logger log.FieldLogger, configPath string) error {
+	logger.Infof("Installing helm chart %s", chart.chartName)
+	if chart.setArgument != "" {
+		cmd := exec.Command("helm", "install", "--kubeconfig", configPath, "--set", chart.setArgument, "-f", chart.valuesPath, chart.chartName, "--namespace", chart.namespace, "--name", chart.chartDeploymentName)
```

Comment:
nit: you can clean up a line here with this instead.
```suggestion
		err := exec.Command("helm", "install", "--kubeconfig", configPath, "--set", chart.setArgument, "-f", chart.valuesPath, chart.chartName, "--namespace", chart.namespace, "--name", chart.chartDeploymentName).Run()
```